dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/proxy/AbstractProxyInvoker.java

```diff
@@ -86,7 +87,7 @@ public Result invoke(Invocation invocation) throws RpcException {
             if (RpcUtils.isReturnTypeFuture(invocation)) {
                 return new AsyncRpcResult((CompletableFuture<Object>) obj);
             } else if (rpcContext.isAsyncStarted()) { // ignore obj in case of RpcContext.startAsync()? always rely on user to write back.
-                return new AsyncRpcResult(rpcContext.getAsyncContext().getInternalFuture());
+                return new AsyncRpcResult(((AsyncContextImpl)(rpcContext.getAsyncContext())).getInternalFuture());
```

Comment:
There could me different implementation of AsyncContext as well ? In this case we might get class cast exception. What do you say? Am I missing something here?

Reply:
@khanimteyaz I think its fine. There's only one impl class and will only have one impl. The main purpose for having to cast here is we don't want to expose 'internal future' to the end user on the interface.